src/main/java/com/example/hbs/controller/BookingController.java

```diff
@@ -14,29 +14,20 @@
     BookingService bookingService;
 
     @RequestMapping(method = RequestMethod.GET, value = "/user/booking/{bookingId}")
-    public ResponseDto<BookingResponseDto> viewBooking(@PathVariable Long bookingId) {
-        BookingResponseDto bookingResponseDto = bookingService.viewBooking(bookingId);
-        if (bookingResponseDto != null)
-            return new ResponseDto<>(bookingResponseDto);
-        else
-            return new ResponseDto<>(null);
+    public ResponseDto<BookingResponseDto> viewBooking(@PathVariable Long bookingId, @RequestParam(defaultValue = "0") Long userId) {
+        BookingResponseDto bookingResponseDto = bookingService.viewBooking(bookingId, userId);
```

Comment:
default value for userId?
check for other places too

Reply:
given in requestparam. Taken "0" as no user can have "0" userId and "0" userId will will return all the hotels

Reply:
this will always throw No such booking Id found, instead we should throw parameter missing.
Ideally, we should not take default value for Id param

Reply:
Okay I got it